src/Data/Functor/Invariant.hs

```diff
@@ -986,3 +986,17 @@ these, you can derive them using the "Data.Functor.Invariant.TH" module.
 genericInvmap :: (Generic1 f, Invariant (Rep1 f)) => (a -> b) -> (b -> a) -> f a -> f b
 genericInvmap f g = to1 . invmap f g . from1
 #endif
+
+-------------------------------------------------------------------------------
+-- Wrappers
+-------------------------------------------------------------------------------
+
+newtype InvariantProfunctor p a = InvariantProfunctor (p a a)
```

Comment:
I'd definitely like the Haddocks for `InvariantProfunctor` to compare it with `WrappedProfunctor`, which serves a similar (but distinct) purpose.